src/main/treeDataProviders/issuesTree/descriptorTree/descriptorTreeNode.ts

```diff
@@ -22,9 +22,11 @@ export class DescriptorTreeNode extends FileTreeNode {
     private _scannedCve?: Set<string> | undefined;
     // Is applicable if key in here
     private _applicableCve?: Map<string, CveApplicableDetails> | undefined;
+    private workspace: string;
 
-    constructor(fileFullPath: string, packageType?: PackageType, parent?: IssuesRootTreeNode) {
-        super(fileFullPath, parent);
+    constructor(filePath: string, packageType?: PackageType, parent?: IssuesRootTreeNode) {
+        super(filePath, parent);
+        this.workspace = filePath.substring(0, filePath.lastIndexOf(path.sep));
```

Comment:
Workspace has a different meaning in VS-Code. Also, I may miss the point, but how can it work if the working directory is not always the first directory in the workspace?
One last thing - if you really want to read the directory name please use the standard methods:
```typescript
path.basename(filePath);
```

Reply:
It is possible that the name is misleading. The update command needs to know where the descriptor is located.
Regarding your suggestion, I think that `path.dirname` is the right method to be replaced with.